aten/src/ATen/native/GatedLinearUnit.cpp

```diff
@@ -3,38 +3,38 @@
 #include <ATen/native/Activation.h>
 
 namespace at {
-namespace native {
-
-// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
-DEFINE_DISPATCH(glu_stub);
-// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
-DEFINE_DISPATCH(glu_backward_stub);
 
-Tensor& glu_out(const Tensor& self, int64_t dim, Tensor &result) {
+namespace meta {
+TORCH_META_FUNC(glu) (
+    const Tensor& self, int64_t dim
+) {
   // this can't pass anyway because a 0-dimensional tensor has "size" 1, which
   // can't be evenly halved, but give a nicer error message here.
   TORCH_CHECK(self.dim() > 0, "glu does not support 0-dimensional tensors");
   auto wrap_dim = maybe_wrap_dim(dim, self.dim());
   const int64_t nIn = self.size(wrap_dim);
   TORCH_CHECK(nIn % 2 == 0, "Halving dimension must be even, but dimension ",
               wrap_dim, " is size ", nIn);
+
   // size output to half of input
   const int64_t selfSize = nIn / 2;
   auto newSizes = self.sizes().vec();
   newSizes[wrap_dim] = selfSize;
```

Comment:
It seems like `newSizes` is not used at all?

Reply:
ah right that was from before (when I was using `set_output`). I'll remove but I would like someone to confirm for me that indeed the substitution `set_output -> build_borrowing_binary_op` is correct.

Reply:
Check if tests pass, but I think this is OK because the narrowed firstHalf/secondHalf tensors will induce TensorIterator to produce the correct size.